Automattic/woocommerce-payments
# Track when platform checkout is offered

Fixes #3829

#### Changes proposed in this Pull Request

On the checkout page, when a customer enters an email address that is not already registered for platform checkout, the customer is presented with a "Remember your details?" section to sign up for platform checkout. This PR adds tracking for when that event takes place.

Worth noting: the section may be offered multiple times. Right now, the event triggers right after we dispatch the `PlatformCheckoutUserCheckEvent` event for a non-registered user. This means that if a customer enters an invalid email address, changes it to a valid email address, then changes it back to an invalid email address, the event will be recorded twice (this will also happen if the customer types their email address too slowly). If that's not desired behavior, I'm open to alternative approaches.

#### Testing instructions

- Enable platform checkout by setting the `_wcpay_feature_platform_checkout` option to `1`.
- Make sure Jetpack ToS are agreed to by setting the `jetpack_tos_agreed` option to `1` if not already set.
- On the merchant site, add a product to cart and go to the checkout page.
- Enter an invalid email into the email field and notice that the "Remember your details?" section appears.
- In Tracks Live View in about 5 minutes, look for a `woocommerceanalytics_platform_checkout_offered` event.
- Enter a valid email into the email field, notice that the "Remember your details?" section disappears.
- In Tracks Live View in about 5 minutes, there shouldn't be a `woocommerceanalytics_platform_checkout_offered` event when the valid email was entered.

-------------------


**Post merge**
